### ISSUE TITLE
Decrypting files fails

### ISSUE THREAD

**Issue by JMBokhorst** [syseleven/nextcloud-tools#27]
Hi, 

I have an issue with decrypting files locally on my server. I was running NC20 but the server went down and now I cannot access the files of one user anymore because they are encrypted. Note other users have the same issue, but they had a full copy of their files on their PC. 

Initially, I tried to decrypt all files with the build-in decrypt function `occ encryption:decrypt-all` however this didn't decrypt any file. This did however result in an error so I cannot access the files anymore via the GUI / Desktop app. 

I tried to use the decrypt-all-files.php script, but that fails on many files, i.e.: 
```
php -f decrypt-nextcloud.php /mnt/storage/restore/ /mnt/storage/nextcloud/jm/files/Photos/weekend-van-de-wetenschap-pathologie.jpg 
DEBUG: debug mode enabled
DEBUG: loaded private key for master_d7e0ad1c
DEBUG: loaded private key for pubShare_d7e0ad1c
DEBUG: filename = /mnt/storage/nextcloud/jm/files/Photos/weekend-van-de-wetenschap-pathologie.jpg
DEBUG: target = /mnt/storage/restore/jm/files/Photos/weekend-van-de-wetenschap-pathologie.jpg
DEBUG: datafilename = Photos/weekend-van-de-wetenschap-pathologie.jpg
DEBUG: istrashbin = false
DEBUG: username = jm
DEBUG: filekeyname = /mnt/storage/nextcloud/jm/files_encryption/keys/files/Photos/weekend-van-de-wetenschap-pathologie.jpg/OC_DEFAULT_MODULE/fileKey
DEBUG: isencrypted = true
DEBUG: secretkey = unavailable
DEBUG: cannot decrypt this file...
DEBUG: success = false
ERROR: /mnt/storage/nextcloud/jm/files/Photos/weekend-van-de-wetenschap-pathologie.jpg FAILED
DEBUG: exiting
```

I see that the secretkey cannot be found, I looked around and see that the file seems to be here: 

```
jm@home:/var/www# ls /mnt/storage/nextcloud/jm/files_encryption/keys/files/Photos/weekend-van-de-wetenschap-pathologie.jpg/OC_DEFAULT_MODULE/
fileKey  jm.shareKey  recoveryKey_d7e0ad1c.shareKey
```

I saw somebody with a similar issue in #21, the problem seems identical but if I rename the file my issue is still identical. 

Hope you can help me.

**Comment by yahesh** [comment]
@JMBokhorst According to the output the following keys could be loaded: `master_d7e0ad1c` and `pubShare_d7e0ad1c`, but the following keys are used to encrypt the actual file: `jm` and `recoveryKey_d7e0ad1c`.

So it seems as if you switched from the master-key encryption to the user-key encryption but you didn't provide the necessary configuration to the script to load the corresponding keys. Specifically have a look at `RECOVERY_PASSWORD` and `USER_PASSWORD_*` (in your case this would be `USER_PASSWORD_JM`).

**Comment by yahesh** [comment]
@JMBokhorst Is this still an issue?

**Comment by JMBokhorst** [comment]
@yahesh, My apologies for the late response! 

I haven't been able to get the script to work. I tried it with the recovery passwords but still had the same problem. 

In the end, I found a way to fix the NX20 version on my machine so I could temporarily access the files via the browser. After I downloaded everything, I started with a new install of Nextcloud, as the original installation started from nextcloud version 11. This might also cause some legacy issues?

For me the problem is solved for now :) Thanks for your time and help
